lib/apiservers/engine/backends/container.go

```diff
@@ -671,7 +671,11 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	} else {
 		state, err := c.containerProxy.State(vc)
 		if err != nil {
-			return err
+			if IsNotFoundError(err) {
+				cache.ContainerCache().DeleteContainer(id)
```

Comment:
We could (and I believe there are similar calls elsewhere), but the reality is we'll get them w/events... _famous last words_